docs/source/reader_and_writer.rst

```diff
@@ -135,6 +135,106 @@ Basically, it's just a high level api on top of the native
     df.write_ext.mysql('localhost', 'my_database', 'my_table',
                        options={'user': 'root', 'password': 'root'})
 
+.. _kafka:
+
+Kafka
+^^^^^
+
+Spark streaming Kafka package (`org.apache.spark:spark-streaming-kafka <https://mvnrepository.com/artifact/org.apache.spark/spark-streaming-kafka-0-10_2.11/2.0.0>`_)
+only provides a basic functionality to represent Kafka topics as RDDs.
+Sparkly provides a more convenient interface for both read and write to Kafka topics.
+This introduces a dependecy on `kafka-python <https://github.com/dpkp/kafka-python>`_ library.
+
++---------------+--------------------------------------------------------------------------------------------+
+| Package       | https://mvnrepository.com/artifact/org.apache.spark/spark-streaming-kafka-0-10_2.11/2.0.0  |
++---------------+--------------------------------------------------------------------------------------------+
+| Configuration | http://spark.apache.org/docs/latest/streaming-kafka-0-10-integration.html                  |
++---------------+--------------------------------------------------------------------------------------------+
+
+Features:
+ - Autodicover topic partitions and offsets.
+ - Work with both RDD and Dataframe for read and write.
+
+.. note::
+    - To use Kafka functionality **sparkly** needs the **kafka-python** library which is an optional dependency.
+      So you need to install **sparkly** with **kafka** extras:
+      ```
+      pip install sparkly[kafka]
+      ```
+    - When working via RDD api, it is expected to be organized as 2-element tuples, which are key and value:
```

Comment:
IMO, we don't need to provide RDD api. The library is designed to work with DataFrames. I can suggest the next protocol to work with Kafka in Sparkly:
1) Reader: in case if user doesn't provide key/value serialisers we can return DataFrame like (key: bytes, value: bytes).
2) Writer: in case if user doesn't provide key/value serialisers we can check that schema is (key: bytes, value: bytes). If not - raise exception.